src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -863,13 +870,23 @@ public static object DeepCopyInner(object original)
             {
                 copy = copier(original);
                 SerializationContext.Current.RecordObject(original, copy);
+                return copy;
             }
-            else
+
+            var externalSerializer = externalSerializers.FirstOrDefault(serializer => serializer.IsSupportedType(t));
```

Comment:
My suspicion is that a single serializer is less error prone, testable, and more comprehensible to developers.

Cases where multiple serialization methods should be conditionally applied (secure messages vs. insecure messages for instance), I would favor a single serializer that maintained both the set of possible serializers to use as well as the logic to decide which to use.

If the default serializer as well as other specific serializers (bond) all conform to the same interfaces, it should not be hard for someone to compose various serializers into a smart serializer that predictably use the correct serializer under specific conditions.  Such conditional serializers would be more reliable and testable than relying on configuration or load order.